src/components/ConnectionStream.tsx

```diff
@@ -0,0 +1,308 @@
+import type { MouseEventHandler } from 'react'
```

Comment:
This file replaces `EngineStream.tsx`. It stores state within the component during the runtime of the file page.

It enables many `src/hooks/network`  hooks. This is the central location in the react land to start connections, disconnect, reconnect, and all of the possible runtime scenarios based on the hooks.